### ISSUE TITLE
Search on extension

### ISSUE THREAD

**Issue by quasarea** [boyter/searchcode-server#160]
Maybe it would be possible to introduce similar to fn parameter for extension, like:
fe - File extension. E.G. fe:cs

I'm fighting with unknown types and it would be great to be able to determine scope of each

**Comment by boyter** [comment]
Certainly it is possible.

So one of the issues with this is that the number of extensions grows as a massive rate. I implemented this on searchcode.com and you run into the situation of what to do with things like Dockerfiles which have no extension. Do you treat the whole thing as an extension? 

What about files with multiple periods in them, for example typings.d.ts

Its something I am looking at doing. What might work better though is that if #158 was implemented and if you could easily see what files are categorised as "Unknown" and easily add them to be classified correctly.

**Comment by quasarea** [comment]
In my case no extantion means really old fortean :) Maybe reverse filename
then. It would allow to search for ts.d.* for such typescript files? Just
late night idea ;)

On 9 Oct 2017 22:01, "Ben Boyter" <notifications@github.com> wrote:

Certainly it is possible.

So one of the issues with this is that the number of extensions grows as a
massive rate. I implemented this on searchcode.com and you run into the
situation of what to do with things like Dockerfiles which have no
extension. Do you treat the whole thing as an extension?

What about files with multiple periods in them, for example typings.d.ts

Its something I am looking at doing. What might work better though is that
if #158 <https://github.com/boyter/searchcode-server/issues/158> was
implemented and if you could easily see what files are categorised as
"Unknown" and easily add them to be classified correctly.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/boyter/searchcode-server/issues/160#issuecomment-335287578>,
or mute the thread

**Comment by boyter** [comment]
Actually the reversed filename is a dammed good idea.

**Comment by boyter** [comment]
Give that a go. In master.

<img width="972" alt="screen shot 2017-10-10 at 5 06 37 pm" src="https://user-images.githubusercontent.com/612151/31371769-7208de90-adde-11e7-9cb0-f14dedac746f.png">

**Comment by quasarea** [comment]
reindexing... 
still I would rather see it like something complementary to fn: functionality
surely working only in filename scope rather then on whole code, 
maybe highlighting reverse results in some way, so you actually that it met reverse condition.
maybe even merging results from both searches, together under certain condition.

if asterisk is in the middle, you know prefix and postfix already, so such search is not needed, 
but it its at the end, it means you don't know one of those. so you could merger results for both

* `filename*` would produce results for `filename.test` and `test.filename`

I will play a bit with this as I may talk nonsense

**Comment by boyter** [comment]
Agreed but I would need a solution to the above. Its something that if I can get a good solution to I will implement.
